ena-submission/scripts/ena_submission_helper.py

```diff
@@ -144,10 +159,266 @@ def get_project_xml(project_set):
+        errors.append(error_message)
+        return CreationResults(results=None, errors=errors, warnings=warnings)
+    try:
+        parsed_response = xmltodict.parse(response.text)
+        valid = (
+            parsed_response["RECEIPT"]["@success"] == "true"
+            and parsed_response["RECEIPT"]["SAMPLE"]["@accession"]
+            and parsed_response["RECEIPT"]["SAMPLE"]["EXT_ID"]["@type"] == "biosample"
+            and parsed_response["RECEIPT"]["SAMPLE"]["EXT_ID"]["@accession"]
+            and parsed_response["RECEIPT"]["SUBMISSION"]["@accession"]
+        )
+        if not valid:
+            raise requests.exceptions.RequestException
+    except:
+        error_message = (
+            f"Response is in unexpected format. "
+            f"Request: {response.request}, Response: {response.text}"
+        )
+        logger.warning(error_message)
+        errors.append(error_message)
+        return CreationResults(results=None, errors=errors, warnings=warnings)
+    sample_results = {
+        "ena_sample_accession": parsed_response["RECEIPT"]["SAMPLE"]["@accession"],
+        "biosample_accession": parsed_response["RECEIPT"]["SAMPLE"]["EXT_ID"]["@accession"],
+        "ena_submission_accession": parsed_response["RECEIPT"]["SUBMISSION"]["@accession"],
+    }
+    return CreationResults(results=sample_results, errors=errors, warnings=warnings)
+
+
 def post_webin(xml, config: ENAConfig):
     return requests.post(
         config.ena_submission_url,
         auth=HTTPBasicAuth(config.ena_submission_username, config.ena_submission_password),
         files=xml,
         timeout=10,  # wait a full 10 seconds for a response incase slow
     )
+
+
+def create_chromosome_list(list_object: AssemblyChromosomeListFile) -> str:
+    """
+    Creates a temp file chromosome list:
+    https://ena-docs.readthedocs.io/en/latest/submit/fileprep/assembly.html#chromosome-list-file
+    """
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".gz") as temp:
+        filename = temp.name
+
+    with gzip.GzipFile(filename, "wb") as gz:
+        for entry in list_object.chromosomes:
+            gz.write(
+                f"{entry.object_name}\t{entry.chromosome_name}\t{entry.topology!s}-{entry.chromosome_type!s}\n".encode()
+            )
+
+    return filename
+
+
+def create_fasta(
+    unaligned_sequences: dict[str, str], chromosome_list: AssemblyChromosomeListFile
+) -> str:
+    """
+    Creates a temp fasta file:
+    https://ena-docs.readthedocs.io/en/latest/submit/fileprep/assembly.html#fasta-file
+    """
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".fasta.gz") as temp:
+        filename = temp.name
+
+    with gzip.GzipFile(filename, "wb") as gz:
+        if len(unaligned_sequences.keys()) == 1:
+            entry = chromosome_list.chromosomes[0]
+            gz.write(f">{entry.object_name}\n".encode())
+            gz.write(f"{unaligned_sequences["main"].replace('-', 'N')}\n".encode())
```

Comment:
Sorry for bringing this up if it has been discussed already and it was decided to be the right way. I'm just a bit surprised to see a gap being replaced by an `N`. Wouldn't it make more sense to simply remove the `-` as they represent emptiness?

Reply:
Or to handle this outside ENA submission - IMO Loculus should never contain raw sequences with `-`s

Reply:
I removed '-' from the example data: https://github.com/pathoplexus/example_data/pull/4 - I will add a prepro check before removing this code.